pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1588,6 +1624,36 @@ struct SlruSegmentDirectory {
     segments: HashSet<u32>,
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+#[repr(u8)]
+pub(crate) enum DirectoryKind {
+    Db,
+    TwoPhase,
+    AuxFiles,
+    SlruSegment(SlruKind),
+}
+
+impl DirectoryKind {
+    pub(crate) const ALL_KINDS: [DirectoryKind; 6] = [
+        DirectoryKind::Db,
+        DirectoryKind::TwoPhase,
+        DirectoryKind::AuxFiles,
+        DirectoryKind::SlruSegment(SlruKind::Clog),
+        DirectoryKind::SlruSegment(SlruKind::MultiXactMembers),
+        DirectoryKind::SlruSegment(SlruKind::MultiXactOffsets),
+    ];
+    pub(crate) fn offset(&self) -> usize {
+        match self {
+            DirectoryKind::Db => 0,
+            DirectoryKind::TwoPhase => 1,
+            DirectoryKind::AuxFiles => 2,
+            DirectoryKind::SlruSegment(SlruKind::Clog) => 3,
+            DirectoryKind::SlruSegment(SlruKind::MultiXactMembers) => 4,
+            DirectoryKind::SlruSegment(SlruKind::MultiXactOffsets) => 5,
```

Comment:
The `enum-set` or `enum-map` used by rest of the metrics could generate this function. Not that I would ever insist on using it, just FYI.

Reply:
Done